pkg/ottl/contexts/ottlprofilestack/profilestack.go

```diff
@@ -0,0 +1,213 @@
+	tc := TransformContext{
+		stack:                stack,
+		sample:               sample,
+		profile:              profile,
+		dictionary:           dictionary,
+		instrumentationScope: instrumentationScope,
+		resource:             resource,
+		cache:                pcommon.NewMap(),
+		scopeProfiles:        scopeProfiles,
+		resourceProfiles:     resourceProfiles,
+	}
+	for _, opt := range options {
+		opt(&tc)
+	}
+	return tc
+}
+
+// GetProfile returns the profile from the TransformContext.
+func (tCtx TransformContext) GetProfile() pprofile.Profile {
+	return tCtx.profile
+}
+
+// GetProfileSample returns the sample from the TransformContext.
+func (tCtx TransformContext) GetProfileSample() pprofile.Sample {
+	return tCtx.sample
+}
+
+// GetProfileStack returns the sample from the TransformContext.
+func (tCtx TransformContext) GetProfileStack() pprofile.Stack {
+	return tCtx.stack
+}
+
+// GetProfilesDictionary returns the profiles dictionary from the TransformContext.
+func (tCtx TransformContext) GetProfilesDictionary() pprofile.ProfilesDictionary {
+	return tCtx.dictionary
+}
+
+// GetInstrumentationScope returns the instrumentation scope from the TransformContext.
+func (tCtx TransformContext) GetInstrumentationScope() pcommon.InstrumentationScope {
+	return tCtx.instrumentationScope
+}
+
+// GetResource returns the resource from the TransformContext.
+func (tCtx TransformContext) GetResource() pcommon.Resource {
+	return tCtx.resource
+}
+
+// GetScopeSchemaURLItem returns the scope schema URL item from the TransformContext.
+func (tCtx TransformContext) GetScopeSchemaURLItem() ctxcommon.SchemaURLItem {
+	return tCtx.scopeProfiles
+}
+
+// GetResourceSchemaURLItem returns the resource schema URL item from the TransformContext.
+func (tCtx TransformContext) GetResourceSchemaURLItem() ctxcommon.SchemaURLItem {
+	return tCtx.resourceProfiles
+}
+
+// NewParser creates a new profile parser with the provided functions and options.
+func NewParser(functions map[string]ottl.Factory[TransformContext], telemetrySettings component.TelemetrySettings, options ...ottl.Option[TransformContext]) (ottl.Parser[TransformContext], error) {
+	return ctxcommon.NewParser(
+		functions,
+		telemetrySettings,
+		pathExpressionParser(getCache),
+		parseEnum,
+		options...,
+	)
+}
+
+// EnablePathContextNames enables the support for path's context names on statements.
+// When this option is configured, all statement's paths must have a valid context prefix,
+// otherwise an error is reported.
+//
+// Experimental: *NOTE* this option is subject to change or removal in the future.
+func EnablePathContextNames() ottl.Option[TransformContext] {
+	return func(p *ottl.Parser[TransformContext]) {
+		ottl.WithPathContextNames[TransformContext]([]string{
+			ctxprofilestack.Name,
```

Comment:
We need to add the upper level contexts here `ctxprofilesample.Name` and `ctxprofile.Name`